text/0000-non-exhaustive.md

```diff
@@ -433,6 +437,41 @@ match it like:
 let Unit { .. } = unit;
 ```
 
+To users of this crate, this will act exactly as if the struct were defined as:
+
+```
+#[non_exhaustive]
+pub struct Unit {}
+```
+
+## Functional record updates
+
+Functional record updates will operate exactly the same regardless of whether
+structs are marked as non-exhaustive or not. For example, given this struct:
```

Comment:
Hm, the decision was the opposite - FRU is not permitted with `non_exhaustive` structs.
FRU (with current rules) is a promise to not add private fields and we want to be able to add private fields to `non_exhaustive` structs.

Reply:
I thought that FRU still worked when structs have private fields?

Reply:
No :(

Reply:
Unless I'm misunderstanding, I tried this out on the playground and it seems to work: https://play.rust-lang.org/?gist=eabe36303f9da0ced44c237ca3b29d45&version=stable

Reply:
https://play.rust-lang.org/?gist=48760b36d43778b49725151313af6be5&version=nightly

Reply:
…right, I'll update the RFC.